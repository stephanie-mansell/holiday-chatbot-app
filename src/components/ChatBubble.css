.bubble {
  border-radius: 15px;

  width: fit-content;
  padding: 10px;
  clear: both;
  margin: 10px 20px;
  font-size: 13px;
}

@keyframes pan-agent {
  from {
    transform: translateX(-100%);
    opacity: 0.1;
  }

  to {
    transform: translateX(0%);
    opacity: 1;
  }
}

@keyframes pan-user {
  from {
    transform: translateX(100%);
    opacity: 0.1;
  }

  to {
    transform: translateX(0%);
    opacity: 1;
  }
}

.user {
  background-color: #f39e91;
  float: right;
  animation: pan-user 0.4s;
}

.agent {
  background-color: #ffb83c;
  float: left;
  animation: pan-agent 0.4s;
}
