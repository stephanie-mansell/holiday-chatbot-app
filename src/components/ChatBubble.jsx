import "./ChatBubble.css";

function ChatBubble(props) {
  return (
    <div
      className={`bubble ${props.message.sender}`}
      data-testid="chatbubble-test"
    >
      <span>{props.message.text}</span>
    </div>
  );
}

export default ChatBubble;
