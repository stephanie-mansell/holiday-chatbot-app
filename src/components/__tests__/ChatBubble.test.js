import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import ChatBubble from "../ChatBubble";

afterEach(() => {
  cleanup();
});

let props = { sender: "agent", text: "I'm an agent" };

test("Should render ChatBubble component", () => {
  render(<ChatBubble message={props} />);
  const chatBubble = screen.getByTestId("chatbubble-test");
  expect(chatBubble).toBeInTheDocument();
});

test("Should show 'agent' as className", () => {
  render(<ChatBubble message={props} />);
  const chatBubble = screen.getByTestId("chatbubble-test");
  expect(chatBubble).toHaveClass("agent");
});

test("Should show 'user' as className", () => {
  props = { sender: "user", text: "I'm a user" };
  render(<ChatBubble message={props} />);
  const chatBubble = screen.getByTestId("chatbubble-test");
  expect(chatBubble).toHaveClass("user");
});

test("Matches snapshot", () => {
  const tree = renderer.create(<ChatBubble message={props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
