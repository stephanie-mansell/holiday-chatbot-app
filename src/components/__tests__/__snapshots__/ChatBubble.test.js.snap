// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="bubble user"
  data-testid="chatbubble-test"
>
  <span>
    I'm a user
  </span>
</div>
`;
